server/src/main/java/org/elasticsearch/http/HttpRouteStats.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.http;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.network.HandlingTimeTracker;
+import org.elasticsearch.common.unit.ByteSizeValue;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.stream.IntStream;
+
+public record HttpRouteStats(
+    long requestCount,
+    long totalRequestSize,
+    long[] requestSizeHistogram,
+    long responseCount,
+    long totalResponseSize,
+    long[] responseSizeHistogram,
+    long[] responseTimeHistogram
```

Comment:
Could we have some javadocs on the meanings of the fields, particularly the histogram buckets?

Reply:
Yep, added in https://github.com/elastic/elasticsearch/pull/99852/commits/5c1fdb781c4513377b779c7c3f6ef0d7586e7b42#diff-1fceaf5362a4c836b7b3b8f6667ce24848028b1fa9f2fdd833a12d3a72df7a1f